pkg/common/terraform/terraform.go

```diff
@@ -0,0 +1,114 @@
+package terraform
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/hc-install/product"
+	"github.com/hashicorp/hc-install/releases"
+	"github.com/hashicorp/terraform-exec/tfexec"
+)
+
+type runner struct {
+	Context    context.Context
+	Installer  *releases.LatestVersion
+	Runner     *tfexec.Terraform
+	WorkingDir string
+}
+
+// New creates a terraform runner for the provided working directory where
+// terraform files are. It returns a TerraformRunner struct which has the
+// runner to perform any commands terraform-exec package provides.
+func New(workingDir string) (*runner, error) {
+	installer := releases.LatestVersion{
+		Product:    product.Terraform,
+		InstallDir: "/tmp",
+	}
+
+	execPath, err := installer.Install(context.Background())
+	if err != nil {
+		return nil, fmt.Errorf("error installing terraform: %w", err)
+	}
+
+	tf, err := tfexec.NewTerraform(workingDir, execPath)
+	if err != nil {
+		return nil, fmt.Errorf("error configuring terraform: %w", err)
+	}
+
+	return &runner{
+		Context:    context.Background(),
```

Comment:
instead of storing a context on the object, prefer taking a context as a parameter to the function it is needed in

Reply:
Resolved in https://github.com/openshift/osde2e/pull/1558/commits/0de34542c4d705ab0513096a30d520fbf86d25bb